### PR TITLE
test: add ft for tidb/issues/51204

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233, close https://github.com/pingcap/tidb/issues/51204

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/fullstack-test/mpp/apply_with_late_materialization.test [hunk 1]
```diff
@@ -78,3 +78,26 @@ mysql> set @@tidb_allow_mpp=0; set @@tidb_allow_tiflash_cop = 1; set @@tidb_allo
...
+mysql> insert into test.t select * from test.t; insert into test.t select * from test.t; insert into test.t select * from test.t;
+mysql> insert into test.t select * from test.t; insert into test.t select * from test.t; insert into test.t select * from test.t;
+mysql> insert into test.t select * from test.t; insert into test.t select * from test.t;
+mysql> create table test.t1 (id int, company_no char(18), name varchar(200), tax_registry_no varchar(30), UNIQUE KEY `uk_company_no` (`company_no`));
+mysql> insert into test.t1 (id, company_no, name, tax_registry_no) values(1, 'Z9005', 'AA', 'aaa'),(2, 'Z9006', 'BB', 'bbb'),(3, 'Z9007', 'CC', 'ccc');
+mysql> alter table test.t set tiflash replica 1;
+
+func> wait_table test t
+
+mysql> select company_no, ifnull((select taxpayer_no from test.t where client_no = c.company_no and client_type = 1 order by update_time desc limit 1), tax_registry_no) as tax_registry_no from test.t1 c where company_no = 'Z9005' limit 1;
```
Comment (SeaRise) [inline tests/fullstack-test/mpp/apply_with_late_materialization.test:94]:
```suggestion
mysql> set tidb_allow_mpp=1;set tidb_enforce_mpp=1; set tidb_isolation_read_engines='tiflash'; select company_no, ifnull((select taxpayer_no from test.t where client_no = c.company_no and client_type = 1 order by update_time desc limit 1), tax_registry_no) as tax_registry_no from test.t1 c where company_no = 'Z9005' limit 1;
```

Comment (Lloyd-Pottiger) [reply tests/fullstack-test/mpp/apply_with_late_materialization.test:94]: Now we will always use MPP, and test.t1 has no tiflash replica.

Comment (SeaRise) [reply tests/fullstack-test/mpp/apply_with_late_materialization.test:94]: ok

**Review by SeaRise — APPROVED** [review]

**Pull request merged**